web/src/stream_list.ts

```diff
@@ -714,7 +716,11 @@ export function refresh_pinned_or_unpinned_stream(sub: StreamSubscription): void
 
 export function refresh_muted_or_unmuted_stream(sub: StreamSubscription): void {
     build_stream_sidebar_row(sub);
-    update_streams_sidebar();
+    // If a stream is inactive, it'll stay in the same inactive list in its
+    // StreamListSection (so `same_as_before` will be false), so we need to
+    // force the rerender to change its faded/unfaded appearance.
+    const force_rerender = !stream_list_sort.has_recent_activity(sub);
+    update_streams_sidebar(force_rerender);
```

Comment:
I think you should drop `build_stream_sidebar_row`?

Reply:
Ended up fully removing this function ([#frontend > removeClass not working? @ 💬](https://chat.zulip.org/#narrow/channel/6-frontend/topic/removeClass.20not.20working.3F/near/2237038))